reactivexcomponent/tests/unit/test_websocket_publisher.py

```diff
@@ -0,0 +1,27 @@
+import unittest
+from mock import Mock
+from reactivexcomponent.communication.publisher import Publisher
+from mock_publisher_dependencies import ReturnMock as Moc
+from reactivexcomponent.configuration.serializer import to_websocket_input_format
+
+
+class TestWebSocketPublisher(unittest.TestCase):
+
+    def setUp(self):
+        self.publisher = Publisher()
+        self.publisher.get_component_code = Moc.Mock["Publisher"].get_component_code
+        self.publisher.get_state_machine_code = Moc.Mock["Publisher"].get_state_machine_code
+        self.publisher.get_publisher_details = Moc.Mock["Publisher"].get_publisher_details
+        self.publisher.websocket = Moc.Mock["create_mock_websocket"]
+
+    def test_send_method(self):
+        self.publisher.websocket.send = Mock()
```

Comment:
there's a problem here... where are we actually testing in this class? self.publisher is a mock, what's the point of testing the behavior of methods in a mock?

The right way of testing it should be:

1. publisher should receive the websocket as parameter in the constructor
2. your test should mock the websocket like you do in L18,
3. your test should check if when we call self.publisher.sender(...) the right calls are done on the web socket (like you do here).